bootstrap.lua

```diff
@@ -0,0 +1,125 @@
+-- This script installs rocks.nvim through a bootstrapping process.
+-- The process consists of the following:
+-- - Configure luarocks to work flawlessly with Neovim
+-- - Install luarocks
+-- - Use the new luarocks installation to install `rocks.nvim`
+
+-- The rocks.nvim plugin is already loaded via the vim.opt.runtimepath:append()
+-- call in the `init.lua` bootstrapping script.
+
+math.randomseed(os.time())
+
+local config_data = vim.g.rocks_nvim or {}
+local install_path = config_data.rocks_path or vim.fs.joinpath(vim.fn.stdpath("data"), "rocks")
+local luarocks_binary = config_data.luarocks_binary or vim.fs.joinpath(install_location, "bin", "luarocks")
+
+---@param dep string
+---@return boolean is_missing
+local function guard_set_up_luarocks_dependency_missing(dep)
+    if vim.fn.executable(dep) ~= 1 then
+        vim.notify(dep .. " must be installed to set up luarocks.", vim.log.levels.ERROR)
+        return true
+    end
+    return false
+end
+
+--- Notify command output.
+---@param msg string
+---@param sc vim.SystemCompleted
+---@param level integer|nil
+local function notify_output(msg, sc, level)
+    local function remove_shell_color(s)
+        return tostring(s):gsub("\x1B%[[0-9;]+m", "")
+    end
+    vim.notify(
+        table.concat({
+            msg,
+            sc and "stderr: " .. remove_shell_color(sc.stderr),
+            sc and "stdout: " .. remove_shell_color(sc.stdout),
+        }, "\n"),
+        level
+    )
+end
+
+--- Sets up luarocks for use with rocks.nvim
+---@param install_path string
+---@return boolean success
+local function set_up_luarocks(install_path)
```

Comment:
issue: this parameter needs to be renamed, as it collides with the top-level variable `install_path` and causes `luacheck` to fail in the CI due to shadowing value.